[Bug] [Worker] Retry send message to master error

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.


### What happened

We found some exception logs on the worker server.

```
[INFO] 2023-08-28 14:55:23.165 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[129] - [WorkflowInstance-0][TaskInstance-954] - Begin retry send message to master, event: TaskInstanceExecutionRunningEvent(taskInstanceId=954, processInstanceId=1545, startTime=1693202790052, host=dolphinscheduler-worker-0.dolphinscheduler-worker-headless:1234, status=TaskExecutionStatus{code=1, desc='running'}, logPath=/opt/dolphinscheduler/logs/20230821/10645509557920/1/1545/954.log, executePath=null, processId=0, appIds=null, eventCreateTime=0, eventSendTime=1693205422750)
[ERROR] 2023-08-28 14:55:23.166 +0800 org.apache.dolphinscheduler.extract.base.server.JdkDynamicServerHandler:[84] - [WorkflowInstance-0][TaskInstance-0] - Cannot find the ServerMethodInvoker of : Transporter(header=TransporterHeader(methodIdentifier=public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionRunning(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionRunningEvent), opaque=22), body=[123, 34, 97, 114, 103, 115, 34, 58, 91, 34, 101, 121, 74, 48, 89, 88, 78, 114, 83, 87, 53, 122, 100, 71, 70, 117, 89, 50, 86, 74, 90, 67, 73, 54, 79, 84, 85, 48, 76, 67, 74, 119, 99, 109, 57, 106, 90, 88, 78, 122, 83, 87, 53, 122, 100, 71, 70, 117, 89, 50, 86, 74, 90, 67, 73, 54, 77, 84, 85, 48, 78, 83, 119, 105, 99, 51, 82, 104, 99, 110, 82, 85, 97, 87, 49, 108, 73, 106, 111, 120, 78, 106, 107, 122, 77, 106, 65, 121, 78, 122, 107, 119, 77, 68, 85, 121, 76, 67, 74, 111, 98, 51, 78, 48, 73, 106, 111, 105, 90, 71, 57, 115, 99, 71, 104, 112, 98, 110, 78, 106, 97, 71, 86, 107, 100, 87, 120, 108, 99, 105, 49, 51, 98, 51, 74, 114, 90, 88, 73, 116, 77, 67, 53, 107, 98, 50, 120, 119, 97, 71, 108, 117, 99, 50, 78, 111, 90, 87, 82, 49, 98, 71, 86, 121, 76, 88, 100, 118, 99, 109, 116, 108, 99, 105, 49, 111, 90, 87, 70, 107, 98, 71, 86, 122, 99, 122, 111, 120, 77, 106, 77, 48, 73, 105, 119, 105, 99, 51, 82, 104, 100, 72, 86, 122, 73, 106, 111, 105, 85, 108, 86, 79, 84, 107, 108, 79, 82, 49, 57, 70, 87, 69, 86, 68, 86, 86, 82, 74, 84, 48, 52, 105, 76, 67, 74, 115, 98, 50, 100, 81, 89, 88, 82, 111, 73, 106, 111, 105, 76, 50, 57, 119, 100, 67, 57, 107, 98, 50, 120, 119, 97, 71, 108, 117, 99, 50, 78, 111, 90, 87, 82, 49, 98, 71, 86, 121, 76, 50, 120, 118, 90, 51, 77, 118, 77, 106, 65, 121, 77, 122, 65, 52, 77, 106, 69, 118, 77, 84, 65, 50, 78, 68, 85, 49, 77, 68, 107, 49, 78, 84, 99, 53, 77, 106, 65, 118, 77, 83, 56, 120, 78, 84, 81, 49, 76, 122, 107, 49, 78, 67, 53, 115, 98, 50, 99, 105, 76, 67, 74, 108, 101, 71, 86, 106, 100, 88, 82, 108, 85, 71, 70, 48, 97, 67, 73, 54, 98, 110, 86, 115, 98, 67, 119, 105, 99, 72, 74, 118, 89, 50, 86, 122, 99, 48, 108, 107, 73, 106, 111, 119, 76, 67, 74, 104, 99, 72, 66, 74, 90, 72, 77, 105, 79, 109, 53, 49, 98, 71, 119, 115, 73, 109, 86, 50, 90, 87, 53, 48, 81, 51, 74, 108, 89, 88, 82, 108, 86, 71, 108, 116, 90, 83, 73, 54, 77, 67, 119, 105, 90, 88, 90, 108, 98, 110, 82, 84, 90, 87, 53, 107, 86, 71, 108, 116, 90, 83, 73, 54, 77, 84, 89, 53, 77, 122, 73, 119, 78, 84, 99, 121, 77, 122, 69, 50, 78, 88, 48, 61, 34, 93, 44, 34, 97, 114, 103, 115, 84, 121, 112, 101, 115, 34, 58, 91, 34, 111, 114, 103, 46, 97, 112, 97, 99, 104, 101, 46, 100, 111, 108, 112, 104, 105, 110, 115, 99, 104, 101, 100, 117, 108, 101, 114, 46, 101, 120, 116, 114, 97, 99, 116, 46, 109, 97, 115, 116, 101, 114, 46, 116, 114, 97, 110, 115, 112, 111, 114, 116, 111, 114, 46, 84, 97, 115, 107, 73, 110, 115, 116, 97, 110, 99, 101, 69, 120, 101, 99, 117, 116, 105, 111, 110, 82, 117, 110, 110, 105, 110, 103, 69, 118, 101, 110, 116, 34, 93, 125])
[WARN] 2023-08-28 14:55:23.167 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[136] - [WorkflowInstance-0][TaskInstance-954] - Retry send message to master error
org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException: Cannot find the ServerMethodInvoker of public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionRunning(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionRunningEvent)
	at org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException.of(MethodInvocationException.java:27)
	at org.apache.dolphinscheduler.extract.base.client.SyncClientMethodInvoker.invoke(SyncClientMethodInvoker.java:48)
	at org.apache.dolphinscheduler.extract.base.client.ClientInvocationHandler.invoke(ClientInvocationHandler.java:55)
	at com.sun.proxy.$Proxy167.onTaskInstanceExecutionRunning(Unknown Source)
	at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionRunningEventSender.sendEvent(TaskInstanceExecutionRunningEventSender.java:41)
	at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionRunningEventSender.sendEvent(TaskInstanceExecutionRunningEventSender.java:30)
	at org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner.run(MessageRetryRunner.java:131)
[INFO] 2023-08-28 14:55:24.168 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[129] - [WorkflowInstance-0][TaskInstance-954] - Begin retry send message to master, event: TaskInstanceExecutionFinishEvent(taskInstanceId=954, processInstanceId=1545, status=6, startTime=1693202790052, host=dolphinscheduler-worker-0.dolphinscheduler-worker-headless:1234, logPath=/opt/dolphinscheduler/logs/20230821/10645509557920/1/1545/954.log, executePath=/tmp/dolphinscheduler/exec/process/default/10579449480160/10645509557920_1/1545/954, endTime=1693202791555, processId=0, appIds=null, varPool=null, eventCreateTime=0, eventSendTime=1693205423753)
[ERROR] 2023-08-28 14:55:24.169 +0800 org.apache.dolphinscheduler.extract.base.server.JdkDynamicServerHandler:[84] - [WorkflowInstance-0][TaskInstance-0] - Cannot find the ServerMethodInvoker of : Transporter(header=TransporterHeader(methodIdentifier=public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionFinish(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionFinishEvent), opaque=23), body=[123, 34, 97, 114, 103, 115, 34, 58, 91, 34, 101, 121, 74, 48, 89, 88, 78, 114, 83, 87, 53, 122, 100, 71, 70, 117, 89, 50, 86, 74, 90, 67, 73, 54, 79, 84, 85, 48, 76, 67, 74, 119, 99, 109, 57, 106, 90, 88, 78, 122, 83, 87, 53, 122, 100, 71, 70, 117, 89, 50, 86, 74, 90, 67, 73, 54, 77, 84, 85, 48, 78, 83, 119, 105, 99, 51, 82, 104, 100, 72, 86, 122, 73, 106, 111, 50, 76, 67, 74, 122, 100, 71, 70, 121, 100, 70, 82, 112, 98, 87, 85, 105, 79, 106, 69, 50, 79, 84, 77, 121, 77, 68, 73, 51, 79, 84, 65, 119, 78, 84, 73, 115, 73, 109, 104, 118, 99, 51, 81, 105, 79, 105, 74, 107, 98, 50, 120, 119, 97, 71, 108, 117, 99, 50, 78, 111, 90, 87, 82, 49, 98, 71, 86, 121, 76, 88, 100, 118, 99, 109, 116, 108, 99, 105, 48, 119, 76, 109, 82, 118, 98, 72, 66, 111, 97, 87, 53, 122, 89, 50, 104, 108, 90, 72, 86, 115, 90, 88, 73, 116, 100, 50, 57, 121, 97, 50, 86, 121, 76, 87, 104, 108, 89, 87, 82, 115, 90, 88, 78, 122, 79, 106, 69, 121, 77, 122, 81, 105, 76, 67, 74, 115, 98, 50, 100, 81, 89, 88, 82, 111, 73, 106, 111, 105, 76, 50, 57, 119, 100, 67, 57, 107, 98, 50, 120, 119, 97, 71, 108, 117, 99, 50, 78, 111, 90, 87, 82, 49, 98, 71, 86, 121, 76, 50, 120, 118, 90, 51, 77, 118, 77, 106, 65, 121, 77, 122, 65, 52, 77, 106, 69, 118, 77, 84, 65, 50, 78, 68, 85, 49, 77, 68, 107, 49, 78, 84, 99, 53, 77, 106, 65, 118, 77, 83, 56, 120, 78, 84, 81, 49, 76, 122, 107, 49, 78, 67, 53, 115, 98, 50, 99, 105, 76, 67, 74, 108, 101, 71, 86, 106, 100, 88, 82, 108, 85, 71, 70, 48, 97, 67, 73, 54, 73, 105, 57, 48, 98, 88, 65, 118, 90, 71, 57, 115, 99, 71, 104, 112, 98, 110, 78, 106, 97, 71, 86, 107, 100, 87, 120, 108, 99, 105, 57, 108, 101, 71, 86, 106, 76, 51, 66, 121, 98, 50, 78, 108, 99, 51, 77, 118, 90, 71, 86, 109, 89, 88, 86, 115, 100, 67, 56, 120, 77, 68, 85, 51, 79, 84, 81, 48, 79, 84, 81, 52, 77, 68, 69, 50, 77, 67, 56, 120, 77, 68, 89, 48, 78, 84, 85, 119, 79, 84, 85, 49, 78, 122, 107, 121, 77, 70, 56, 120, 76, 122, 69, 49, 78, 68, 85, 118, 79, 84, 85, 48, 73, 105, 119, 105, 90, 87, 53, 107, 86, 71, 108, 116, 90, 83, 73, 54, 77, 84, 89, 53, 77, 122, 73, 119, 77, 106, 99, 53, 77, 84, 85, 49, 78, 83, 119, 105, 99, 72, 74, 118, 89, 50, 86, 122, 99, 48, 108, 107, 73, 106, 111, 119, 76, 67, 74, 104, 99, 72, 66, 74, 90, 72, 77, 105, 79, 109, 53, 49, 98, 71, 119, 115, 73, 110, 90, 104, 99, 108, 66, 118, 98, 50, 119, 105, 79, 109, 53, 49, 98, 71, 119, 115, 73, 109, 86, 50, 90, 87, 53, 48, 81, 51, 74, 108, 89, 88, 82, 108, 86, 71, 108, 116, 90, 83, 73, 54, 77, 67, 119, 105, 90, 88, 90, 108, 98, 110, 82, 84, 90, 87, 53, 107, 86, 71, 108, 116, 90, 83, 73, 54, 77, 84, 89, 53, 77, 122, 73, 119, 78, 84, 99, 121, 78, 68, 69, 50, 79, 72, 48, 61, 34, 93, 44, 34, 97, 114, 103, 115, 84, 121, 112, 101, 115, 34, 58, 91, 34, 111, 114, 103, 46, 97, 112, 97, 99, 104, 101, 46, 100, 111, 108, 112, 104, 105, 110, 115, 99, 104, 101, 100, 117, 108, 101, 114, 46, 101, 120, 116, 114, 97, 99, 116, 46, 109, 97, 115, 116, 101, 114, 46, 116, 114, 97, 110, 115, 112, 111, 114, 116, 111, 114, 46, 84, 97, 115, 107, 73, 110, 115, 116, 97, 110, 99, 101, 69, 120, 101, 99, 117, 116, 105, 111, 110, 70, 105, 110, 105, 115, 104, 69, 118, 101, 110, 116, 34, 93, 125])
[WARN] 2023-08-28 14:55:24.171 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[136] - [WorkflowInstance-0][TaskInstance-954] - Retry send message to master error
org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException: Cannot find the ServerMethodInvoker of public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionFinish(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionFinishEvent)
	at org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException.of(MethodInvocationException.java:27)
	at org.apache.dolphinscheduler.extract.base.client.SyncClientMethodInvoker.invoke(SyncClientMethodInvoker.java:48)
	at org.apache.dolphinscheduler.extract.base.client.ClientInvocationHandler.invoke(ClientInvocationHandler.java:55)
	at com.sun.proxy.$Proxy167.onTaskInstanceExecutionFinish(Unknown Source)
	at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionFinishEventSender.sendEvent(TaskInstanceExecutionFinishEventSender.java:39)
	at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionFinishEventSender.sendEvent(TaskInstanceExecutionFinishEventSender.java:28)
	at org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner.run(MessageRetryRunner.java:131)
```

### What you expected to happen

no error

### How to reproduce

A new cluster has been deployed through k8s with the dev branch, and MySQL is used as the database.

server: 1 master, 1 worker.

Both the master and worker were restarted at the same time, and then the worker produced the above log. These logs will continue to be generated repeatedly.

@ruanwenjun @Radeity @EricGao888  PTAL.

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

```
[ERROR] 2023-08-28 17:22:56.647 +0800 org.apache.dolphinscheduler.extract.base.server.JdkDynamicServerHandler:[84] - [WorkflowInstance-0][TaskInstance-0] - Cannot find the ServerMethodInvoker of : Transporter(header=TransporterHeader(methodIdentifier=public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionInfoUpdate(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionInfoEvent), opaque=2), body=[123, 34, 97, 114, 103, 115, 34, 58, 91, 34, 101, 121, 74, 48, 89, 88, 78, 114, 83, 87, 53, 122, 100, 71, 70, 117, 89, 50, 86, 74, 90, 67, 73, 54, 77, 121, 119, 105, 99, 72, 74, 118, 89, 50, 86, 122, 99, 48, 108, 117, 99, 51, 82, 104, 98, 109, 78, 108, 83, 87, 81, 105, 79, 106, 73, 115, 73, 110, 78, 48, 89, 88, 74, 48, 86, 71, 108, 116, 90, 83, 73, 54, 77, 84, 89, 53, 77, 122, 73, 120, 78, 68, 85, 51, 78, 106, 85, 121, 78, 67, 119, 105, 97, 71, 57, 122, 100, 67, 73, 54, 73, 106, 69, 119, 76, 106, 69, 53, 77, 105, 52, 52, 78, 121, 52, 121, 77, 106, 69, 54, 77, 84, 73, 122, 78, 67, 73, 115, 73, 109, 120, 118, 90, 49, 66, 104, 100, 71, 103, 105, 79, 109, 53, 49, 98, 71, 119, 115, 73, 110, 66, 121, 98, 50, 78, 108, 99, 51, 78, 74, 90, 67, 73, 54, 77, 67, 119, 105, 90, 88, 90, 108, 98, 110, 82, 68, 99, 109, 86, 104, 100, 71, 86, 85, 97, 87, 49, 108, 73, 106, 111, 119, 76, 67, 74, 108, 100, 109, 86, 117, 100, 70, 78, 108, 98, 109, 82, 85, 97, 87, 49, 108, 73, 106, 111, 119, 102, 81, 61, 61, 34, 93, 44, 34, 97, 114, 103, 115, 84, 121, 112, 101, 115, 34, 58, 91, 34, 111, 114, 103, 46, 97, 112, 97, 99, 104, 101, 46, 100, 111, 108, 112, 104, 105, 110, 115, 99, 104, 101, 100, 117, 108, 101, 114, 46, 101, 120, 116, 114, 97, 99, 116, 46, 109, 97, 115, 116, 101, 114, 46, 116, 114, 97, 110, 115, 112, 111, 114, 116, 111, 114, 46, 84, 97, 115, 107, 73, 110, 115, 116, 97, 110, 99, 101, 69, 120, 101, 99, 117, 116, 105, 111, 110, 73, 110, 102, 111, 69, 118, 101, 110, 116, 34, 93, 125])
[ERROR] 2023-08-28 17:22:56.648 +0800 org.apache.dolphinscheduler.server.worker.rpc.WorkerMessageSender:[68] - [WorkflowInstance-2][TaskInstance-3] - Send message error, eventType: RUNNING_INFO, event: TaskInstanceExecutionInfoEvent(taskInstanceId=3, processInstanceId=2, startTime=1693214576524, host=10.192.87.221:1234, logPath=null, processId=0, eventCreateTime=0, eventSendTime=0)

```

A new cluster was created, and a new workflow was created to execute a shell task that runs 'echo 123;'. The logs show that the worker has run successfully, but an error occurred on the worker. According to the API server, the task is in the dispatching state.

To view the logs of the task, an exception similar to issue [14819](https://github.com/apache/dolphinscheduler/issues/14819) occurred. The following is the exception log on the worker.


```
[ERROR] 2023-08-28 17:25:05.005 +0800 org.apache.dolphinscheduler.extract.base.server.JdkDynamicServerHandler:[115] - [WorkflowInstance-0][TaskInstance-0] - Invoke method public abstract org.apache.dolphinscheduler.extract.worker.transportor.TaskInstanceLogPageQueryResponse org.apache.dolphinscheduler.extract.worker.IWorkerLogService.pageQueryTaskInstanceLog(org.apache.dolphinscheduler.extract.worker.transportor.TaskInstanceLogPageQueryRequest) failed, null.
java.lang.reflect.InvocationTargetException: null
	at sun.reflect.GeneratedMethodAccessor112.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.dolphinscheduler.extract.base.server.ServerMethodInvokerImpl.invoke(ServerMethodInvokerImpl.java:39)
	at org.apache.dolphinscheduler.extract.base.server.JdkDynamicServerHandler.lambda$processReceived$0(JdkDynamicServerHandler.java:108)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException: null
	at java.io.File.<init>(File.java:277)
	at org.apache.dolphinscheduler.common.utils.LogUtils.readPartFileContent(LogUtils.java:103)
	at org.apache.dolphinscheduler.server.worker.rpc.WorkerLogServiceImpl.pageQueryTaskInstanceLog(WorkerLogServiceImpl.java:56)
	... 11 common frames omitted
```

me too


```bash
[ERROR] 2023-08-28 17:42:55.229 +0800 org.apache.dolphinscheduler.extract.base.server.JdkDynamicServerHandler:[115] - [WorkflowInstance-0][TaskInstance-0] - Invoke method public abstract org.apache.dolphinscheduler.extract.worker.transportor.TaskInstanceLogPageQueryResponse org.apache.dolphinscheduler.extract.worker.IWorkerLogService.pageQueryTaskInstanceLog(org.apache.dolphinscheduler.extract.worker.transportor.TaskInstanceLogPageQueryRequest) failed, null.
java.lang.reflect.InvocationTargetException: null
        at sun.reflect.GeneratedMethodAccessor105.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.dolphinscheduler.extract.base.server.ServerMethodInvokerImpl.invoke(ServerMethodInvokerImpl.java:39)
        at org.apache.dolphinscheduler.extract.base.server.JdkDynamicServerHandler.lambda$processReceived$0(JdkDynamicServerHandler.java:108)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:750)
Caused by: java.lang.NullPointerException: null
        at java.io.File.<init>(File.java:279)
        at org.apache.dolphinscheduler.common.utils.LogUtils.readPartFileContent(LogUtils.java:103)
        at org.apache.dolphinscheduler.server.worker.rpc.WorkerLogServiceImpl.pageQueryTaskInstanceLog(WorkerLogServiceImpl.java:56)
        ... 10 common frames omitted
```




```bash
[ERROR] 2023-08-28 18:21:16.824 +0800 org.apache.dolphinscheduler.extract.base.server.JdkDynamicServerHandler:[115] - [WorkflowInstance-0][TaskInstance-0] - Invoke method public abstract org.apache.dolphinscheduler.extract.worker.transportor.TaskInstanceLogPageQueryResponse org.apache.dolphinscheduler.extract.worker.IWorkerLogService.pageQueryTaskInstanceLog(org.apache.dolphinscheduler.extract.worker.transportor.TaskInstanceLogPageQueryRequest) failed, null.
java.lang.reflect.InvocationTargetException: null
        at sun.reflect.GeneratedMethodAccessor70.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.dolphinscheduler.extract.base.server.ServerMethodInvokerImpl.invoke(ServerMethodInvokerImpl.java:39)
        at org.apache.dolphinscheduler.extract.base.server.JdkDynamicServerHandler.lambda$processReceived$0(JdkDynamicServerHandler.java:108)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:750)
Caused by: java.lang.NullPointerException: null
        at java.io.File.<init>(File.java:279)
        at org.apache.dolphinscheduler.common.utils.LogUtils.readPartFileContent(LogUtils.java:103)
        at org.apache.dolphinscheduler.server.worker.rpc.WorkerLogServiceImpl.pageQueryTaskInstanceLog(WorkerLogServiceImpl.java:56)
        ... 10 common frames omitted
[INFO] 2023-08-28 18:21:17.710 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[129] - [WorkflowInstance-0][TaskInstance-3] - Begin retry send message to master, event: TaskInstanceExecutionRunningEvent(taskInstanceId=3, processInstanceId=2, startTime=1693218007975, host=dolphinscheduler-worker-2.dolphinscheduler-worker-headless:1234, status=TaskExecutionStatus{code=1, desc='running'}, logPath=/opt/dolphinscheduler/logs/20230828/10724266729984/1/2/3.log, executePath=null, processId=0, appIds=null, eventCreateTime=0, eventSendTime=0)
[ERROR] 2023-08-28 18:21:17.711 +0800 org.apache.dolphinscheduler.extract.base.server.JdkDynamicServerHandler:[84] - [WorkflowInstance-0][TaskInstance-0] - Cannot find the ServerMethodInvoker of : Transporter(header=TransporterHeader(methodIdentifier=public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionRunning(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionRunningEvent), opaque=5), body=[123, 34, 97, 114, 103, 115, 34, 58, 91, 34, 101, 121, 74, 48, 89, 88, 78, 114, 83, 87, 53, 122, 100, 71, 70, 117, 89, 50, 86, 74, 90, 67, 73, 54, 77, 121, 119, 105, 99, 72, 74, 118, 89, 50, 86, 122, 99, 48, 108, 117, 99, 51, 82, 104, 98, 109, 78, 108, 83, 87, 81, 105, 79, 106, 73, 115, 73, 110, 78, 48, 89, 88, 74, 48, 86, 71, 108, 116, 90, 83, 73, 54, 77, 84, 89, 53, 77, 122, 73, 120, 79, 68, 65, 119, 78, 122, 107, 51, 78, 83, 119, 105, 97, 71, 57, 122, 100, 67, 73, 54, 73, 109, 82, 118, 98, 72, 66, 111, 97, 87, 53, 122, 89, 50, 104, 108, 90, 72, 86, 115, 90, 88, 73, 116, 100, 50, 57, 121, 97, 50, 86, 121, 76, 84, 73, 117, 90, 71, 57, 115, 99, 71, 104, 112, 98, 110, 78, 106, 97, 71, 86, 107, 100, 87, 120, 108, 99, 105, 49, 51, 98, 51, 74, 114, 90, 88, 73, 116, 97, 71, 86, 104, 90, 71, 120, 108, 99, 51, 77, 54, 77, 84, 73, 122, 78, 67, 73, 115, 73, 110, 78, 48, 89, 88, 82, 49, 99, 121, 73, 54, 73, 108, 74, 86, 84, 107, 53, 74, 84, 107, 100, 102, 82, 86, 104, 70, 81, 49, 86, 85, 83, 85, 57, 79, 73, 105, 119, 105, 98, 71, 57, 110, 85, 71, 70, 48, 97, 67, 73, 54, 73, 105, 57, 118, 99, 72, 81, 118, 90, 71, 57, 115, 99, 71, 104, 112, 98, 110, 78, 106, 97, 71, 86, 107, 100, 87, 120, 108, 99, 105, 57, 115, 98, 50, 100, 122, 76, 122, 73, 119, 77, 106, 77, 119, 79, 68, 73, 52, 76, 122, 69, 119, 78, 122, 73, 48, 77, 106, 89, 50, 78, 122, 73, 53, 79, 84, 103, 48, 76, 122, 69, 118, 77, 105, 56, 122, 76, 109, 120, 118, 90, 121, 73, 115, 73, 109, 86, 52, 90, 87, 78, 49, 100, 71, 86, 81, 89, 88, 82, 111, 73, 106, 112, 117, 100, 87, 120, 115, 76, 67, 74, 119, 99, 109, 57, 106, 90, 88, 78, 122, 83, 87, 81, 105, 79, 106, 65, 115, 73, 109, 70, 119, 99, 69, 108, 107, 99, 121, 73, 54, 98, 110, 86, 115, 98, 67, 119, 105, 90, 88, 90, 108, 98, 110, 82, 68, 99, 109, 86, 104, 100, 71, 86, 85, 97, 87, 49, 108, 73, 106, 111, 119, 76, 67, 74, 108, 100, 109, 86, 117, 100, 70, 78, 108, 98, 109, 82, 85, 97, 87, 49, 108, 73, 106, 111, 120, 78, 106, 107, 122, 77, 106, 69, 52, 77, 68, 99, 51, 78, 122, 69, 119, 102, 81, 61, 61, 34, 93, 44, 34, 97, 114, 103, 115, 84, 121, 112, 101, 115, 34, 58, 91, 34, 111, 114, 103, 46, 97, 112, 97, 99, 104, 101, 46, 100, 111, 108, 112, 104, 105, 110, 115, 99, 104, 101, 100, 117, 108, 101, 114, 46, 101, 120, 116, 114, 97, 99, 116, 46, 109, 97, 115, 116, 101, 114, 46, 116, 114, 97, 110, 115, 112, 111, 114, 116, 111, 114, 46, 84, 97, 115, 107, 73, 110, 115, 116, 97, 110, 99, 101, 69, 120, 101, 99, 117, 116, 105, 111, 110, 82, 117, 110, 110, 105, 110, 103, 69, 118, 101, 110, 116, 34, 93, 125])
[WARN] 2023-08-28 18:21:17.712 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[136] - [WorkflowInstance-0][TaskInstance-3] - Retry send message to master error
org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException: Cannot find the ServerMethodInvoker of public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionRunning(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionRunningEvent)
        at org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException.of(MethodInvocationException.java:27)
        at org.apache.dolphinscheduler.extract.base.client.SyncClientMethodInvoker.invoke(SyncClientMethodInvoker.java:48)
        at org.apache.dolphinscheduler.extract.base.client.ClientInvocationHandler.invoke(ClientInvocationHandler.java:55)
        at com.sun.proxy.$Proxy167.onTaskInstanceExecutionRunning(Unknown Source)
        at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionRunningEventSender.sendEvent(TaskInstanceExecutionRunningEventSender.java:41)
        at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionRunningEventSender.sendEvent(TaskInstanceExecutionRunningEventSender.java:30)
        at org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner.run(MessageRetryRunner.java:131)
[INFO] 2023-08-28 18:21:18.712 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[129] - [WorkflowInstance-0][TaskInstance-3] - Begin retry send message to master, event: TaskInstanceExecutionInfoEvent(taskInstanceId=3, processInstanceId=2, startTime=1693218007975, host=dolphinscheduler-worker-2.dolphinscheduler-worker-headless:1234, logPath=null, processId=0, eventCreateTime=0, eventSendTime=0)
[ERROR] 2023-08-28 18:21:18.714 +0800 org.apache.dolphinscheduler.extract.base.server.JdkDynamicServerHandler:[84] - [WorkflowInstance-0][TaskInstance-0] - Cannot find the ServerMethodInvoker of : Transporter(header=TransporterHeader(methodIdentifier=public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionInfoUpdate(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionInfoEvent), opaque=6), body=[123, 34, 97, 114, 103, 115, 34, 58, 91, 34, 101, 121, 74, 48, 89, 88, 78, 114, 83, 87, 53, 122, 100, 71, 70, 117, 89, 50, 86, 74, 90, 67, 73, 54, 77, 121, 119, 105, 99, 72, 74, 118, 89, 50, 86, 122, 99, 48, 108, 117, 99, 51, 82, 104, 98, 109, 78, 108, 83, 87, 81, 105, 79, 106, 73, 115, 73, 110, 78, 48, 89, 88, 74, 48, 86, 71, 108, 116, 90, 83, 73, 54, 77, 84, 89, 53, 77, 122, 73, 120, 79, 68, 65, 119, 78, 122, 107, 51, 78, 83, 119, 105, 97, 71, 57, 122, 100, 67, 73, 54, 73, 109, 82, 118, 98, 72, 66, 111, 97, 87, 53, 122, 89, 50, 104, 108, 90, 72, 86, 115, 90, 88, 73, 116, 100, 50, 57, 121, 97, 50, 86, 121, 76, 84, 73, 117, 90, 71, 57, 115, 99, 71, 104, 112, 98, 110, 78, 106, 97, 71, 86, 107, 100, 87, 120, 108, 99, 105, 49, 51, 98, 51, 74, 114, 90, 88, 73, 116, 97, 71, 86, 104, 90, 71, 120, 108, 99, 51, 77, 54, 77, 84, 73, 122, 78, 67, 73, 115, 73, 109, 120, 118, 90, 49, 66, 104, 100, 71, 103, 105, 79, 109, 53, 49, 98, 71, 119, 115, 73, 110, 66, 121, 98, 50, 78, 108, 99, 51, 78, 74, 90, 67, 73, 54, 77, 67, 119, 105, 90, 88, 90, 108, 98, 110, 82, 68, 99, 109, 86, 104, 100, 71, 86, 85, 97, 87, 49, 108, 73, 106, 111, 119, 76, 67, 74, 108, 100, 109, 86, 117, 100, 70, 78, 108, 98, 109, 82, 85, 97, 87, 49, 108, 73, 106, 111, 120, 78, 106, 107, 122, 77, 106, 69, 52, 77, 68, 99, 52, 78, 122, 69, 121, 102, 81, 61, 61, 34, 93, 44, 34, 97, 114, 103, 115, 84, 121, 112, 101, 115, 34, 58, 91, 34, 111, 114, 103, 46, 97, 112, 97, 99, 104, 101, 46, 100, 111, 108, 112, 104, 105, 110, 115, 99, 104, 101, 100, 117, 108, 101, 114, 46, 101, 120, 116, 114, 97, 99, 116, 46, 109, 97, 115, 116, 101, 114, 46, 116, 114, 97, 110, 115, 112, 111, 114, 116, 111, 114, 46, 84, 97, 115, 107, 73, 110, 115, 116, 97, 110, 99, 101, 69, 120, 101, 99, 117, 116, 105, 111, 110, 73, 110, 102, 111, 69, 118, 101, 110, 116, 34, 93, 125])
[WARN] 2023-08-28 18:21:18.714 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[136] - [WorkflowInstance-0][TaskInstance-3] - Retry send message to master error
org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException: Cannot find the ServerMethodInvoker of public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionInfoUpdate(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionInfoEvent)
        at org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException.of(MethodInvocationException.java:27)
        at org.apache.dolphinscheduler.extract.base.client.SyncClientMethodInvoker.invoke(SyncClientMethodInvoker.java:48)
        at org.apache.dolphinscheduler.extract.base.client.ClientInvocationHandler.invoke(ClientInvocationHandler.java:55)
        at com.sun.proxy.$Proxy167.onTaskInstanceExecutionInfoUpdate(Unknown Source)
        at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionInfoUpdateEventSender.sendEvent(TaskInstanceExecutionInfoUpdateEventSender.java:41)
        at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionInfoUpdateEventSender.sendEvent(TaskInstanceExecutionInfoUpdateEventSender.java:30)
        at org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner.run(MessageRetryRunner.java:131)
[INFO] 2023-08-28 18:21:19.714 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[129] - [WorkflowInstance-0][TaskInstance-3] - Begin retry send message to master, event: TaskInstanceExecutionFinishEvent(taskInstanceId=3, processInstanceId=2, status=7, startTime=1693218007975, host=dolphinscheduler-worker-2.dolphinscheduler-worker-headless:1234, logPath=/opt/dolphinscheduler/logs/20230828/10724266729984/1/2/3.log, executePath=/tmp/dolphinscheduler/exec/process/default/10724225436160/10724266729984_1/2/3, endTime=1693218009085, processId=340, appIds=null, varPool=[], eventCreateTime=0, eventSendTime=0)
[ERROR] 2023-08-28 18:21:19.716 +0800 org.apache.dolphinscheduler.extract.base.server.JdkDynamicServerHandler:[84] - [WorkflowInstance-0][TaskInstance-0] - Cannot find the ServerMethodInvoker of : Transporter(header=TransporterHeader(methodIdentifier=public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionFinish(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionFinishEvent), opaque=7), body=[123, 34, 97, 114, 103, 115, 34, 58, 91, 34, 101, 121, 74, 48, 89, 88, 78, 114, 83, 87, 53, 122, 100, 71, 70, 117, 89, 50, 86, 74, 90, 67, 73, 54, 77, 121, 119, 105, 99, 72, 74, 118, 89, 50, 86, 122, 99, 48, 108, 117, 99, 51, 82, 104, 98, 109, 78, 108, 83, 87, 81, 105, 79, 106, 73, 115, 73, 110, 78, 48, 89, 88, 82, 49, 99, 121, 73, 54, 78, 121, 119, 105, 99, 51, 82, 104, 99, 110, 82, 85, 97, 87, 49, 108, 73, 106, 111, 120, 78, 106, 107, 122, 77, 106, 69, 52, 77, 68, 65, 51, 79, 84, 99, 49, 76, 67, 74, 111, 98, 51, 78, 48, 73, 106, 111, 105, 90, 71, 57, 115, 99, 71, 104, 112, 98, 110, 78, 106, 97, 71, 86, 107, 100, 87, 120, 108, 99, 105, 49, 51, 98, 51, 74, 114, 90, 88, 73, 116, 77, 105, 53, 107, 98, 50, 120, 119, 97, 71, 108, 117, 99, 50, 78, 111, 90, 87, 82, 49, 98, 71, 86, 121, 76, 88, 100, 118, 99, 109, 116, 108, 99, 105, 49, 111, 90, 87, 70, 107, 98, 71, 86, 122, 99, 122, 111, 120, 77, 106, 77, 48, 73, 105, 119, 105, 98, 71, 57, 110, 85, 71, 70, 48, 97, 67, 73, 54, 73, 105, 57, 118, 99, 72, 81, 118, 90, 71, 57, 115, 99, 71, 104, 112, 98, 110, 78, 106, 97, 71, 86, 107, 100, 87, 120, 108, 99, 105, 57, 115, 98, 50, 100, 122, 76, 122, 73, 119, 77, 106, 77, 119, 79, 68, 73, 52, 76, 122, 69, 119, 78, 122, 73, 48, 77, 106, 89, 50, 78, 122, 73, 53, 79, 84, 103, 48, 76, 122, 69, 118, 77, 105, 56, 122, 76, 109, 120, 118, 90, 121, 73, 115, 73, 109, 86, 52, 90, 87, 78, 49, 100, 71, 86, 81, 89, 88, 82, 111, 73, 106, 111, 105, 76, 51, 82, 116, 99, 67, 57, 107, 98, 50, 120, 119, 97, 71, 108, 117, 99, 50, 78, 111, 90, 87, 82, 49, 98, 71, 86, 121, 76, 50, 86, 52, 90, 87, 77, 118, 99, 72, 74, 118, 89, 50, 86, 122, 99, 121, 57, 107, 90, 87, 90, 104, 100, 87, 120, 48, 76, 122, 69, 119, 78, 122, 73, 48, 77, 106, 73, 49, 78, 68, 77, 50, 77, 84, 89, 119, 76, 122, 69, 119, 78, 122, 73, 48, 77, 106, 89, 50, 78, 122, 73, 53, 79, 84, 103, 48, 88, 122, 69, 118, 77, 105, 56, 122, 73, 105, 119, 105, 90, 87, 53, 107, 86, 71, 108, 116, 90, 83, 73, 54, 77, 84, 89, 53, 77, 122, 73, 120, 79, 68, 65, 119, 79, 84, 65, 52, 78, 83, 119, 105, 99, 72, 74, 118, 89, 50, 86, 122, 99, 48, 108, 107, 73, 106, 111, 122, 78, 68, 65, 115, 73, 109, 70, 119, 99, 69, 108, 107, 99, 121, 73, 54, 98, 110, 86, 115, 98, 67, 119, 105, 100, 109, 70, 121, 85, 71, 57, 118, 98, 67, 73, 54, 73, 108, 116, 100, 73, 105, 119, 105, 90, 88, 90, 108, 98, 110, 82, 68, 99, 109, 86, 104, 100, 71, 86, 85, 97, 87, 49, 108, 73, 106, 111, 119, 76, 67, 74, 108, 100, 109, 86, 117, 100, 70, 78, 108, 98, 109, 82, 85, 97, 87, 49, 108, 73, 106, 111, 120, 78, 106, 107, 122, 77, 106, 69, 52, 77, 68, 99, 53, 78, 122, 69, 48, 102, 81, 61, 61, 34, 93, 44, 34, 97, 114, 103, 115, 84, 121, 112, 101, 115, 34, 58, 91, 34, 111, 114, 103, 46, 97, 112, 97, 99, 104, 101, 46, 100, 111, 108, 112, 104, 105, 110, 115, 99, 104, 101, 100, 117, 108, 101, 114, 46, 101, 120, 116, 114, 97, 99, 116, 46, 109, 97, 115, 116, 101, 114, 46, 116, 114, 97, 110, 115, 112, 111, 114, 116, 111, 114, 46, 84, 97, 115, 107, 73, 110, 115, 116, 97, 110, 99, 101, 69, 120, 101, 99, 117, 116, 105, 111, 110, 70, 105, 110, 105, 115, 104, 69, 118, 101, 110, 116, 34, 93, 125])
[WARN] 2023-08-28 18:21:19.716 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[136] - [WorkflowInstance-0][TaskInstance-3] - Retry send message to master error
org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException: Cannot find the ServerMethodInvoker of public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionFinish(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionFinishEvent)
        at org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException.of(MethodInvocationException.java:27)
        at org.apache.dolphinscheduler.extract.base.client.SyncClientMethodInvoker.invoke(SyncClientMethodInvoker.java:48)
        at org.apache.dolphinscheduler.extract.base.client.ClientInvocationHandler.invoke(ClientInvocationHandler.java:55)
        at com.sun.proxy.$Proxy167.onTaskInstanceExecutionFinish(Unknown Source)
        at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionFinishEventSender.sendEvent(TaskInstanceExecutionFinishEventSender.java:39)
        at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionFinishEventSender.sendEvent(TaskInstanceExecutionFinishEventSender.java:28)
        at org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner.run(MessageRetryRunner.java:131)
[INFO] 2023-08-28 18:24:17.419 +0800 org.apache.dolphinscheduler.server.worker.runner.operator.TaskInstanceKillOperationFunction:[55] - [WorkflowInstance-0][TaskInstance-0] - Receive TaskInstanceKillRequest: TaskInstanceKillRequest(taskInstanceId=3)
[ERROR] 2023-08-28 18:24:17.419 +0800 org.apache.dolphinscheduler.server.worker.runner.operator.TaskInstanceKillOperationFunction:[63] - [WorkflowInstance-0][TaskInstance-3] - Cannot find TaskExecutionContext for taskInstance: 3
[INFO] 2023-08-28 18:25:51.945 +0800 io.kubernetes.client.informer.cache.ReflectorRunnable:[244] - [WorkflowInstance-0][TaskInstance-0] - ResourceVersion 863489912 and Watch connection expired: too old resource version: 863489912 (863497534) , will retry w/o resourceVersion next time
[INFO] 2023-08-28 18:25:51.945 +0800 io.kubernetes.client.informer.cache.ReflectorRunnable:[139] - [WorkflowInstance-0][TaskInstance-0] - class io.kubernetes.client.openapi.models.V1ConfigMap#Watch expired, will re-list-watch soon
[INFO] 2023-08-28 18:25:52.945 +0800 io.kubernetes.client.informer.cache.ReflectorRunnable:[84] - [WorkflowInstance-0][TaskInstance-0] - class io.kubernetes.client.openapi.models.V1ConfigMap#Start listing and watching...
[INFO] 2023-08-28 18:25:53.006 +0800 o.s.cloud.kubernetes.client.config.reload.KubernetesClientEventBasedConfigMapChangeDetector:[123] - [WorkflowInstance-0][TaskInstance-0] - ConfigMap common-conf was added.
[WARN] 2023-08-28 18:25:53.011 +0800 o.s.cloud.kubernetes.client.config.reload.KubernetesClientEventBasedConfigMapChangeDetector:[151] - [WorkflowInstance-0][TaskInstance-0] - Configuration change was not detected.
[INFO] 2023-08-28 18:25:53.011 +0800 o.s.cloud.kubernetes.client.config.reload.KubernetesClientEventBasedConfigMapChangeDetector:[123] - [WorkflowInstance-0][TaskInstance-0] - ConfigMap scripts was added.
[WARN] 2023-08-28 18:25:53.014 +0800 o.s.cloud.kubernetes.client.config.reload.KubernetesClientEventBasedConfigMapChangeDetector:[151] - [WorkflowInstance-0][TaskInstance-0] - Configuration change was not detected.
[INFO] 2023-08-28 18:26:17.752 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[129] - [WorkflowInstance-0][TaskInstance-3] - Begin retry send message to master, event: TaskInstanceExecutionRunningEvent(taskInstanceId=3, processInstanceId=2, startTime=1693218007975, host=dolphinscheduler-worker-2.dolphinscheduler-worker-headless:1234, status=TaskExecutionStatus{code=1, desc='running'}, logPath=/opt/dolphinscheduler/logs/20230828/10724266729984/1/2/3.log, executePath=null, processId=0, appIds=null, eventCreateTime=0, eventSendTime=1693218077710)
[ERROR] 2023-08-28 18:26:17.754 +0800 org.apache.dolphinscheduler.extract.base.server.JdkDynamicServerHandler:[84] - [WorkflowInstance-0][TaskInstance-0] - Cannot find the ServerMethodInvoker of : Transporter(header=TransporterHeader(methodIdentifier=public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionRunning(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionRunningEvent), opaque=8), body=[123, 34, 97, 114, 103, 115, 34, 58, 91, 34, 101, 121, 74, 48, 89, 88, 78, 114, 83, 87, 53, 122, 100, 71, 70, 117, 89, 50, 86, 74, 90, 67, 73, 54, 77, 121, 119, 105, 99, 72, 74, 118, 89, 50, 86, 122, 99, 48, 108, 117, 99, 51, 82, 104, 98, 109, 78, 108, 83, 87, 81, 105, 79, 106, 73, 115, 73, 110, 78, 48, 89, 88, 74, 48, 86, 71, 108, 116, 90, 83, 73, 54, 77, 84, 89, 53, 77, 122, 73, 120, 79, 68, 65, 119, 78, 122, 107, 51, 78, 83, 119, 105, 97, 71, 57, 122, 100, 67, 73, 54, 73, 109, 82, 118, 98, 72, 66, 111, 97, 87, 53, 122, 89, 50, 104, 108, 90, 72, 86, 115, 90, 88, 73, 116, 100, 50, 57, 121, 97, 50, 86, 121, 76, 84, 73, 117, 90, 71, 57, 115, 99, 71, 104, 112, 98, 110, 78, 106, 97, 71, 86, 107, 100, 87, 120, 108, 99, 105, 49, 51, 98, 51, 74, 114, 90, 88, 73, 116, 97, 71, 86, 104, 90, 71, 120, 108, 99, 51, 77, 54, 77, 84, 73, 122, 78, 67, 73, 115, 73, 110, 78, 48, 89, 88, 82, 49, 99, 121, 73, 54, 73, 108, 74, 86, 84, 107, 53, 74, 84, 107, 100, 102, 82, 86, 104, 70, 81, 49, 86, 85, 83, 85, 57, 79, 73, 105, 119, 105, 98, 71, 57, 110, 85, 71, 70, 48, 97, 67, 73, 54, 73, 105, 57, 118, 99, 72, 81, 118, 90, 71, 57, 115, 99, 71, 104, 112, 98, 110, 78, 106, 97, 71, 86, 107, 100, 87, 120, 108, 99, 105, 57, 115, 98, 50, 100, 122, 76, 122, 73, 119, 77, 106, 77, 119, 79, 68, 73, 52, 76, 122, 69, 119, 78, 122, 73, 48, 77, 106, 89, 50, 78, 122, 73, 53, 79, 84, 103, 48, 76, 122, 69, 118, 77, 105, 56, 122, 76, 109, 120, 118, 90, 121, 73, 115, 73, 109, 86, 52, 90, 87, 78, 49, 100, 71, 86, 81, 89, 88, 82, 111, 73, 106, 112, 117, 100, 87, 120, 115, 76, 67, 74, 119, 99, 109, 57, 106, 90, 88, 78, 122, 83, 87, 81, 105, 79, 106, 65, 115, 73, 109, 70, 119, 99, 69, 108, 107, 99, 121, 73, 54, 98, 110, 86, 115, 98, 67, 119, 105, 90, 88, 90, 108, 98, 110, 82, 68, 99, 109, 86, 104, 100, 71, 86, 85, 97, 87, 49, 108, 73, 106, 111, 119, 76, 67, 74, 108, 100, 109, 86, 117, 100, 70, 78, 108, 98, 109, 82, 85, 97, 87, 49, 108, 73, 106, 111, 120, 78, 106, 107, 122, 77, 106, 69, 52, 77, 122, 99, 51, 78, 122, 85, 121, 102, 81, 61, 61, 34, 93, 44, 34, 97, 114, 103, 115, 84, 121, 112, 101, 115, 34, 58, 91, 34, 111, 114, 103, 46, 97, 112, 97, 99, 104, 101, 46, 100, 111, 108, 112, 104, 105, 110, 115, 99, 104, 101, 100, 117, 108, 101, 114, 46, 101, 120, 116, 114, 97, 99, 116, 46, 109, 97, 115, 116, 101, 114, 46, 116, 114, 97, 110, 115, 112, 111, 114, 116, 111, 114, 46, 84, 97, 115, 107, 73, 110, 115, 116, 97, 110, 99, 101, 69, 120, 101, 99, 117, 116, 105, 111, 110, 82, 117, 110, 110, 105, 110, 103, 69, 118, 101, 110, 116, 34, 93, 125])
[WARN] 2023-08-28 18:26:17.755 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[136] - [WorkflowInstance-0][TaskInstance-3] - Retry send message to master error
org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException: Cannot find the ServerMethodInvoker of public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionRunning(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionRunningEvent)
        at org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException.of(MethodInvocationException.java:27)
        at org.apache.dolphinscheduler.extract.base.client.SyncClientMethodInvoker.invoke(SyncClientMethodInvoker.java:48)
        at org.apache.dolphinscheduler.extract.base.client.ClientInvocationHandler.invoke(ClientInvocationHandler.java:55)
        at com.sun.proxy.$Proxy167.onTaskInstanceExecutionRunning(Unknown Source)
        at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionRunningEventSender.sendEvent(TaskInstanceExecutionRunningEventSender.java:41)
        at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionRunningEventSender.sendEvent(TaskInstanceExecutionRunningEventSender.java:30)
        at org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner.run(MessageRetryRunner.java:131)
[INFO] 2023-08-28 18:26:18.755 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[129] - [WorkflowInstance-0][TaskInstance-3] - Begin retry send message to master, event: TaskInstanceExecutionInfoEvent(taskInstanceId=3, processInstanceId=2, startTime=1693218007975, host=dolphinscheduler-worker-2.dolphinscheduler-worker-headless:1234, logPath=null, processId=0, eventCreateTime=0, eventSendTime=1693218078712)
[ERROR] 2023-08-28 18:26:18.756 +0800 org.apache.dolphinscheduler.extract.base.server.JdkDynamicServerHandler:[84] - [WorkflowInstance-0][TaskInstance-0] - Cannot find the ServerMethodInvoker of : Transporter(header=TransporterHeader(methodIdentifier=public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionInfoUpdate(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionInfoEvent), opaque=9), body=[123, 34, 97, 114, 103, 115, 34, 58, 91, 34, 101, 121, 74, 48, 89, 88, 78, 114, 83, 87, 53, 122, 100, 71, 70, 117, 89, 50, 86, 74, 90, 67, 73, 54, 77, 121, 119, 105, 99, 72, 74, 118, 89, 50, 86, 122, 99, 48, 108, 117, 99, 51, 82, 104, 98, 109, 78, 108, 83, 87, 81, 105, 79, 106, 73, 115, 73, 110, 78, 48, 89, 88, 74, 48, 86, 71, 108, 116, 90, 83, 73, 54, 77, 84, 89, 53, 77, 122, 73, 120, 79, 68, 65, 119, 78, 122, 107, 51, 78, 83, 119, 105, 97, 71, 57, 122, 100, 67, 73, 54, 73, 109, 82, 118, 98, 72, 66, 111, 97, 87, 53, 122, 89, 50, 104, 108, 90, 72, 86, 115, 90, 88, 73, 116, 100, 50, 57, 121, 97, 50, 86, 121, 76, 84, 73, 117, 90, 71, 57, 115, 99, 71, 104, 112, 98, 110, 78, 106, 97, 71, 86, 107, 100, 87, 120, 108, 99, 105, 49, 51, 98, 51, 74, 114, 90, 88, 73, 116, 97, 71, 86, 104, 90, 71, 120, 108, 99, 51, 77, 54, 77, 84, 73, 122, 78, 67, 73, 115, 73, 109, 120, 118, 90, 49, 66, 104, 100, 71, 103, 105, 79, 109, 53, 49, 98, 71, 119, 115, 73, 110, 66, 121, 98, 50, 78, 108, 99, 51, 78, 74, 90, 67, 73, 54, 77, 67, 119, 105, 90, 88, 90, 108, 98, 110, 82, 68, 99, 109, 86, 104, 100, 71, 86, 85, 97, 87, 49, 108, 73, 106, 111, 119, 76, 67, 74, 108, 100, 109, 86, 117, 100, 70, 78, 108, 98, 109, 82, 85, 97, 87, 49, 108, 73, 106, 111, 120, 78, 106, 107, 122, 77, 106, 69, 52, 77, 122, 99, 52, 78, 122, 85, 49, 102, 81, 61, 61, 34, 93, 44, 34, 97, 114, 103, 115, 84, 121, 112, 101, 115, 34, 58, 91, 34, 111, 114, 103, 46, 97, 112, 97, 99, 104, 101, 46, 100, 111, 108, 112, 104, 105, 110, 115, 99, 104, 101, 100, 117, 108, 101, 114, 46, 101, 120, 116, 114, 97, 99, 116, 46, 109, 97, 115, 116, 101, 114, 46, 116, 114, 97, 110, 115, 112, 111, 114, 116, 111, 114, 46, 84, 97, 115, 107, 73, 110, 115, 116, 97, 110, 99, 101, 69, 120, 101, 99, 117, 116, 105, 111, 110, 73, 110, 102, 111, 69, 118, 101, 110, 116, 34, 93, 125])
[WARN] 2023-08-28 18:26:18.756 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[136] - [WorkflowInstance-0][TaskInstance-3] - Retry send message to master error
org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException: Cannot find the ServerMethodInvoker of public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionInfoUpdate(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionInfoEvent)
        at org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException.of(MethodInvocationException.java:27)
        at org.apache.dolphinscheduler.extract.base.client.SyncClientMethodInvoker.invoke(SyncClientMethodInvoker.java:48)
        at org.apache.dolphinscheduler.extract.base.client.ClientInvocationHandler.invoke(ClientInvocationHandler.java:55)
        at com.sun.proxy.$Proxy167.onTaskInstanceExecutionInfoUpdate(Unknown Source)
        at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionInfoUpdateEventSender.sendEvent(TaskInstanceExecutionInfoUpdateEventSender.java:41)
        at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionInfoUpdateEventSender.sendEvent(TaskInstanceExecutionInfoUpdateEventSender.java:30)
        at org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner.run(MessageRetryRunner.java:131)
[INFO] 2023-08-28 18:26:19.757 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[129] - [WorkflowInstance-0][TaskInstance-3] - Begin retry send message to master, event: TaskInstanceExecutionFinishEvent(taskInstanceId=3, processInstanceId=2, status=7, startTime=1693218007975, host=dolphinscheduler-worker-2.dolphinscheduler-worker-headless:1234, logPath=/opt/dolphinscheduler/logs/20230828/10724266729984/1/2/3.log, executePath=/tmp/dolphinscheduler/exec/process/default/10724225436160/10724266729984_1/2/3, endTime=1693218009085, processId=340, appIds=null, varPool=[], eventCreateTime=0, eventSendTime=1693218079714)
[ERROR] 2023-08-28 18:26:19.758 +0800 org.apache.dolphinscheduler.extract.base.server.JdkDynamicServerHandler:[84] - [WorkflowInstance-0][TaskInstance-0] - Cannot find the ServerMethodInvoker of : Transporter(header=TransporterHeader(methodIdentifier=public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionFinish(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionFinishEvent), opaque=10), body=[123, 34, 97, 114, 103, 115, 34, 58, 91, 34, 101, 121, 74, 48, 89, 88, 78, 114, 83, 87, 53, 122, 100, 71, 70, 117, 89, 50, 86, 74, 90, 67, 73, 54, 77, 121, 119, 105, 99, 72, 74, 118, 89, 50, 86, 122, 99, 48, 108, 117, 99, 51, 82, 104, 98, 109, 78, 108, 83, 87, 81, 105, 79, 106, 73, 115, 73, 110, 78, 48, 89, 88, 82, 49, 99, 121, 73, 54, 78, 121, 119, 105, 99, 51, 82, 104, 99, 110, 82, 85, 97, 87, 49, 108, 73, 106, 111, 120, 78, 106, 107, 122, 77, 106, 69, 52, 77, 68, 65, 51, 79, 84, 99, 49, 76, 67, 74, 111, 98, 51, 78, 48, 73, 106, 111, 105, 90, 71, 57, 115, 99, 71, 104, 112, 98, 110, 78, 106, 97, 71, 86, 107, 100, 87, 120, 108, 99, 105, 49, 51, 98, 51, 74, 114, 90, 88, 73, 116, 77, 105, 53, 107, 98, 50, 120, 119, 97, 71, 108, 117, 99, 50, 78, 111, 90, 87, 82, 49, 98, 71, 86, 121, 76, 88, 100, 118, 99, 109, 116, 108, 99, 105, 49, 111, 90, 87, 70, 107, 98, 71, 86, 122, 99, 122, 111, 120, 77, 106, 77, 48, 73, 105, 119, 105, 98, 71, 57, 110, 85, 71, 70, 48, 97, 67, 73, 54, 73, 105, 57, 118, 99, 72, 81, 118, 90, 71, 57, 115, 99, 71, 104, 112, 98, 110, 78, 106, 97, 71, 86, 107, 100, 87, 120, 108, 99, 105, 57, 115, 98, 50, 100, 122, 76, 122, 73, 119, 77, 106, 77, 119, 79, 68, 73, 52, 76, 122, 69, 119, 78, 122, 73, 48, 77, 106, 89, 50, 78, 122, 73, 53, 79, 84, 103, 48, 76, 122, 69, 118, 77, 105, 56, 122, 76, 109, 120, 118, 90, 121, 73, 115, 73, 109, 86, 52, 90, 87, 78, 49, 100, 71, 86, 81, 89, 88, 82, 111, 73, 106, 111, 105, 76, 51, 82, 116, 99, 67, 57, 107, 98, 50, 120, 119, 97, 71, 108, 117, 99, 50, 78, 111, 90, 87, 82, 49, 98, 71, 86, 121, 76, 50, 86, 52, 90, 87, 77, 118, 99, 72, 74, 118, 89, 50, 86, 122, 99, 121, 57, 107, 90, 87, 90, 104, 100, 87, 120, 48, 76, 122, 69, 119, 78, 122, 73, 48, 77, 106, 73, 49, 78, 68, 77, 50, 77, 84, 89, 119, 76, 122, 69, 119, 78, 122, 73, 48, 77, 106, 89, 50, 78, 122, 73, 53, 79, 84, 103, 48, 88, 122, 69, 118, 77, 105, 56, 122, 73, 105, 119, 105, 90, 87, 53, 107, 86, 71, 108, 116, 90, 83, 73, 54, 77, 84, 89, 53, 77, 122, 73, 120, 79, 68, 65, 119, 79, 84, 65, 52, 78, 83, 119, 105, 99, 72, 74, 118, 89, 50, 86, 122, 99, 48, 108, 107, 73, 106, 111, 122, 78, 68, 65, 115, 73, 109, 70, 119, 99, 69, 108, 107, 99, 121, 73, 54, 98, 110, 86, 115, 98, 67, 119, 105, 100, 109, 70, 121, 85, 71, 57, 118, 98, 67, 73, 54, 73, 108, 116, 100, 73, 105, 119, 105, 90, 88, 90, 108, 98, 110, 82, 68, 99, 109, 86, 104, 100, 71, 86, 85, 97, 87, 49, 108, 73, 106, 111, 119, 76, 67, 74, 108, 100, 109, 86, 117, 100, 70, 78, 108, 98, 109, 82, 85, 97, 87, 49, 108, 73, 106, 111, 120, 78, 106, 107, 122, 77, 106, 69, 52, 77, 122, 99, 53, 78, 122, 85, 51, 102, 81, 61, 61, 34, 93, 44, 34, 97, 114, 103, 115, 84, 121, 112, 101, 115, 34, 58, 91, 34, 111, 114, 103, 46, 97, 112, 97, 99, 104, 101, 46, 100, 111, 108, 112, 104, 105, 110, 115, 99, 104, 101, 100, 117, 108, 101, 114, 46, 101, 120, 116, 114, 97, 99, 116, 46, 109, 97, 115, 116, 101, 114, 46, 116, 114, 97, 110, 115, 112, 111, 114, 116, 111, 114, 46, 84, 97, 115, 107, 73, 110, 115, 116, 97, 110, 99, 101, 69, 120, 101, 99, 117, 116, 105, 111, 110, 70, 105, 110, 105, 115, 104, 69, 118, 101, 110, 116, 34, 93, 125])
[WARN] 2023-08-28 18:26:19.758 +0800 org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner:[136] - [WorkflowInstance-0][TaskInstance-3] - Retry send message to master error
org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException: Cannot find the ServerMethodInvoker of public abstract void org.apache.dolphinscheduler.extract.master.ITaskInstanceExecutionEventListener.onTaskInstanceExecutionFinish(org.apache.dolphinscheduler.extract.master.transportor.TaskInstanceExecutionFinishEvent)
        at org.apache.dolphinscheduler.extract.base.exception.MethodInvocationException.of(MethodInvocationException.java:27)
        at org.apache.dolphinscheduler.extract.base.client.SyncClientMethodInvoker.invoke(SyncClientMethodInvoker.java:48)
        at org.apache.dolphinscheduler.extract.base.client.ClientInvocationHandler.invoke(ClientInvocationHandler.java:55)
        at com.sun.proxy.$Proxy167.onTaskInstanceExecutionFinish(Unknown Source)
        at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionFinishEventSender.sendEvent(TaskInstanceExecutionFinishEventSender.java:39)
        at org.apache.dolphinscheduler.server.worker.message.TaskInstanceExecutionFinishEventSender.sendEvent(TaskInstanceExecutionFinishEventSender.java:28)
        at org.apache.dolphinscheduler.server.worker.message.MessageRetryRunner.run(MessageRetryRunner.java:131)

```

--- a/dolphinscheduler-extract/dolphinscheduler-extract-master/src/main/java/org/apache/dolphinscheduler/extract/master/transportor/ITaskInstanceExecutionEvent.java
+++ b/dolphinscheduler-extract/dolphinscheduler-extract-master/src/main/java/org/apache/dolphinscheduler/extract/master/transportor/ITaskInstanceExecutionEvent.java
@@ -27,7 +27,13 @@ public interface ITaskInstanceExecutionEvent {
 
     void setEventSendTime(long eventSendTime);
 
-    void setHost(String host);
+    void setWorkflowInstanceHost(String host);
+
+    String getWorkflowInstanceHost();
+
+    void setTaskInstanceHost(String host);
+
+    String getTaskInstanceHost();
 
     TaskInstanceExecutionEventType getEventType();
 

--- a/dolphinscheduler-extract/dolphinscheduler-extract-master/src/main/java/org/apache/dolphinscheduler/extract/master/transportor/TaskInstanceExecutionFinishEvent.java
+++ b/dolphinscheduler-extract/dolphinscheduler-extract-master/src/main/java/org/apache/dolphinscheduler/extract/master/transportor/TaskInstanceExecutionFinishEvent.java
@@ -34,7 +34,9 @@ public class TaskInstanceExecutionFinishEvent implements ITaskInstanceExecutionE
 
     private long startTime;
 
-    private String host;
+    private String taskInstanceHost;
+
+    private String workflowInstanceHost;
 
     private String logPath;
 

--- a/dolphinscheduler-extract/dolphinscheduler-extract-master/src/main/java/org/apache/dolphinscheduler/extract/master/transportor/TaskInstanceExecutionInfoEvent.java
+++ b/dolphinscheduler-extract/dolphinscheduler-extract-master/src/main/java/org/apache/dolphinscheduler/extract/master/transportor/TaskInstanceExecutionInfoEvent.java
@@ -32,7 +32,9 @@ public class TaskInstanceExecutionInfoEvent implements ITaskInstanceExecutionEve
 
     private long startTime;
 
-    private String host;
+    private String workflowInstanceHost;
+
+    private String taskInstanceHost;
 
     private String logPath;
 

--- a/dolphinscheduler-extract/dolphinscheduler-extract-master/src/main/java/org/apache/dolphinscheduler/extract/master/transportor/TaskInstanceExecutionRunningEvent.java
+++ b/dolphinscheduler-extract/dolphinscheduler-extract-master/src/main/java/org/apache/dolphinscheduler/extract/master/transportor/TaskInstanceExecutionRunningEvent.java
@@ -34,7 +34,9 @@ public class TaskInstanceExecutionRunningEvent implements ITaskInstanceExecution
 
     private long startTime;
 
-    private String host;
+    private String taskInstanceHost;
+
+    private String workflowInstanceHost;
 
     private TaskExecutionStatus status;
 

--- a/dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskEvent.java
+++ b/dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskEvent.java
@@ -114,7 +114,7 @@ public class TaskEvent {
         event.setExecutePath(command.getExecutePath());
         event.setLogPath(command.getLogPath());
         event.setAppIds(command.getAppIds());
-        event.setWorkerAddress(command.getHost());
+        event.setWorkerAddress(command.getTaskInstanceHost());
         event.setEvent(TaskEventType.RUNNING);
         return event;
     }
@@ -131,7 +131,7 @@ public class TaskEvent {
         event.setProcessId(command.getProcessId());
         event.setAppIds(command.getAppIds());
         event.setVarPool(command.getVarPool());
-        event.setWorkerAddress(command.getHost());
+        event.setWorkerAddress(command.getTaskInstanceHost());
         event.setEvent(TaskEventType.RESULT);
         return event;
     }
@@ -151,7 +151,7 @@ public class TaskEvent {
         event.setTaskInstanceId(command.getTaskInstanceId());
         event.setStartTime(DateUtils.timeStampToDate(command.getStartTime()));
         event.setLogPath(command.getLogPath());
-        event.setWorkerAddress(command.getHost());
+        event.setWorkerAddress(command.getTaskInstanceHost());
         event.setEvent(TaskEventType.UPDATE_PID);
         return event;
     }

--- a/dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/message/LogicTaskInstanceExecuteRunningEventSender.java
+++ b/dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/message/LogicTaskInstanceExecuteRunningEventSender.java
@@ -35,7 +35,7 @@ public class LogicTaskInstanceExecuteRunningEventSender
     public void sendMessage(TaskInstanceExecutionRunningEvent taskInstanceExecutionRunningEvent) {
         ITaskInstanceExecutionEventListener iTaskInstanceExecutionEventListener =
                 SingletonJdkDynamicRpcClientProxyFactory
-                        .getProxyClient(taskInstanceExecutionRunningEvent.getHost(),
+                        .getProxyClient(taskInstanceExecutionRunningEvent.getWorkflowInstanceHost(),
                                 ITaskInstanceExecutionEventListener.class);
         iTaskInstanceExecutionEventListener.onTaskInstanceExecutionRunning(taskInstanceExecutionRunningEvent);
     }
@@ -47,7 +47,8 @@ public class LogicTaskInstanceExecuteRunningEventSender
         taskExecuteRunningMessage.setProcessInstanceId(taskExecutionContext.getProcessInstanceId());
         taskExecuteRunningMessage.setStatus(taskExecutionContext.getCurrentExecutionStatus());
         taskExecuteRunningMessage.setLogPath(taskExecutionContext.getLogPath());
-        taskExecuteRunningMessage.setHost(taskExecutionContext.getHost());
+        taskExecuteRunningMessage.setWorkflowInstanceHost(taskExecutionContext.getWorkflowInstanceHost());
+        taskExecuteRunningMessage.setTaskInstanceHost(taskExecutionContext.getHost());
         taskExecuteRunningMessage.setStartTime(taskExecutionContext.getStartTime());
         taskExecuteRunningMessage.setExecutePath(taskExecutionContext.getExecutePath());
         taskExecuteRunningMessage.setAppIds(taskExecutionContext.getAppIds());

--- a/dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/message/LogicTaskInstanceExecutionFinishEventSender.java
+++ b/dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/message/LogicTaskInstanceExecutionFinishEventSender.java
@@ -33,7 +33,7 @@ public class LogicTaskInstanceExecutionFinishEventSender
     public void sendMessage(TaskInstanceExecutionFinishEvent message) {
         ITaskInstanceExecutionEventListener iTaskInstanceExecutionEventListener =
                 SingletonJdkDynamicRpcClientProxyFactory
-                        .getProxyClient(message.getHost(), ITaskInstanceExecutionEventListener.class);
+                        .getProxyClient(message.getWorkflowInstanceHost(), ITaskInstanceExecutionEventListener.class);
         iTaskInstanceExecutionEventListener.onTaskInstanceExecutionFinish(message);
     }
 
@@ -47,7 +47,8 @@ public class LogicTaskInstanceExecutionFinishEventSender
         taskExecuteResultMessage.setExecutePath(taskExecutionContext.getExecutePath());
         taskExecuteResultMessage.setAppIds(taskExecutionContext.getAppIds());
         taskExecuteResultMessage.setProcessId(taskExecutionContext.getProcessId());
-        taskExecuteResultMessage.setHost(taskExecutionContext.getHost());
+        taskExecuteResultMessage.setWorkflowInstanceHost(taskExecutionContext.getWorkflowInstanceHost());
+        taskExecuteResultMessage.setTaskInstanceHost(taskExecutionContext.getHost());
         taskExecuteResultMessage.setStartTime(taskExecutionContext.getStartTime());
         taskExecuteResultMessage.setEndTime(taskExecutionContext.getEndTime());
         taskExecuteResultMessage.setVarPool(taskExecutionContext.getVarPool());

--- a/dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/message/MessageRetryRunner.java
+++ b/dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/message/MessageRetryRunner.java
@@ -96,7 +96,7 @@ public class MessageRetryRunner extends BaseDaemonThread {
         List<TaskInstanceMessage> taskInstanceMessages = this.needToRetryMessages.get(taskInstanceId);
         if (taskInstanceMessages != null) {
             taskInstanceMessages.forEach(taskInstanceMessage -> {
-                taskInstanceMessage.getEvent().setHost(messageReceiverHost);
+                taskInstanceMessage.getEvent().setWorkflowInstanceHost(messageReceiverHost);
             });
         }
     }

--- a/dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/message/TaskInstanceExecutionFinishEventSender.java
+++ b/dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/message/TaskInstanceExecutionFinishEventSender.java
@@ -34,7 +34,7 @@ public class TaskInstanceExecutionFinishEventSender
     public void sendEvent(TaskInstanceExecutionFinishEvent taskInstanceExecutionFinishEvent) {
         ITaskInstanceExecutionEventListener iTaskInstanceExecutionEventListener =
                 SingletonJdkDynamicRpcClientProxyFactory
-                        .getProxyClient(taskInstanceExecutionFinishEvent.getHost(),
+                        .getProxyClient(taskInstanceExecutionFinishEvent.getWorkflowInstanceHost(),
                                 ITaskInstanceExecutionEventListener.class);
         iTaskInstanceExecutionEventListener.onTaskInstanceExecutionFinish(taskInstanceExecutionFinishEvent);
     }
@@ -49,7 +49,8 @@ public class TaskInstanceExecutionFinishEventSender
         taskExecuteResultMessage.setExecutePath(taskExecutionContext.getExecutePath());
         taskExecuteResultMessage.setAppIds(taskExecutionContext.getAppIds());
         taskExecuteResultMessage.setProcessId(taskExecutionContext.getProcessId());
-        taskExecuteResultMessage.setHost(taskExecutionContext.getHost());
+        taskExecuteResultMessage.setWorkflowInstanceHost(taskExecutionContext.getWorkflowInstanceHost());
+        taskExecuteResultMessage.setTaskInstanceHost(taskExecutionContext.getHost());
         taskExecuteResultMessage.setStartTime(taskExecutionContext.getStartTime());
         taskExecuteResultMessage.setEndTime(taskExecutionContext.getEndTime());
         taskExecuteResultMessage.setVarPool(taskExecutionContext.getVarPool());

--- a/dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/message/TaskInstanceExecutionInfoUpdateEventSender.java
+++ b/dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/message/TaskInstanceExecutionInfoUpdateEventSender.java
@@ -36,7 +36,7 @@ public class TaskInstanceExecutionInfoUpdateEventSender
     public void sendEvent(TaskInstanceExecutionInfoEvent taskInstanceExecutionInfoEvent) {
         ITaskInstanceExecutionEventListener iTaskInstanceExecutionEventListener =
                 SingletonJdkDynamicRpcClientProxyFactory
-                        .getProxyClient(taskInstanceExecutionInfoEvent.getHost(),
+                        .getProxyClient(taskInstanceExecutionInfoEvent.getWorkflowInstanceHost(),
                                 ITaskInstanceExecutionEventListener.class);
         iTaskInstanceExecutionEventListener.onTaskInstanceExecutionInfoUpdate(taskInstanceExecutionInfoEvent);
     }
@@ -46,7 +46,8 @@ public class TaskInstanceExecutionInfoUpdateEventSender
         TaskInstanceExecutionInfoEvent taskUpdatePidRequest = new TaskInstanceExecutionInfoEvent();
         taskUpdatePidRequest.setTaskInstanceId(taskExecutionContext.getTaskInstanceId());
         taskUpdatePidRequest.setProcessInstanceId(taskExecutionContext.getProcessInstanceId());
-        taskUpdatePidRequest.setHost(taskExecutionContext.getHost());
+        taskUpdatePidRequest.setWorkflowInstanceHost(taskExecutionContext.getWorkflowInstanceHost());
+        taskUpdatePidRequest.setTaskInstanceHost(taskExecutionContext.getHost());
         taskUpdatePidRequest.setStartTime(taskExecutionContext.getStartTime());
         return taskUpdatePidRequest;
     }

--- a/dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/message/TaskInstanceExecutionRunningEventSender.java
+++ b/dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/message/TaskInstanceExecutionRunningEventSender.java
@@ -36,7 +36,7 @@ public class TaskInstanceExecutionRunningEventSender
     public void sendEvent(TaskInstanceExecutionRunningEvent taskInstanceExecutionRunningEvent) {
         ITaskInstanceExecutionEventListener iTaskInstanceExecutionEventListener =
                 SingletonJdkDynamicRpcClientProxyFactory
-                        .getProxyClient(taskInstanceExecutionRunningEvent.getHost(),
+                        .getProxyClient(taskInstanceExecutionRunningEvent.getWorkflowInstanceHost(),
                                 ITaskInstanceExecutionEventListener.class);
         iTaskInstanceExecutionEventListener.onTaskInstanceExecutionRunning(taskInstanceExecutionRunningEvent);
     }
@@ -48,7 +48,8 @@ public class TaskInstanceExecutionRunningEventSender
         taskExecuteRunningMessage.setProcessInstanceId(taskExecutionContext.getProcessInstanceId());
         taskExecuteRunningMessage.setStatus(taskExecutionContext.getCurrentExecutionStatus());
         taskExecuteRunningMessage.setLogPath(taskExecutionContext.getLogPath());
-        taskExecuteRunningMessage.setHost(taskExecutionContext.getHost());
+        taskExecuteRunningMessage.setWorkflowInstanceHost(taskExecutionContext.getWorkflowInstanceHost());
+        taskExecuteRunningMessage.setTaskInstanceHost(taskExecutionContext.getHost());
         taskExecuteRunningMessage.setStartTime(taskExecutionContext.getStartTime());
         taskExecuteRunningMessage.setExecutePath(taskExecutionContext.getExecutePath());
         taskExecuteRunningMessage.setAppIds(taskExecutionContext.getAppIds());

